backend/src/db/factories.js

```diff
@@ -0,0 +1,226 @@
+import uuid from 'uuid/v4'
+import faker from 'faker'
+import slugify from 'slug'
+import { hashSync } from 'bcryptjs'
+import { Factory } from 'rosie'
+import { getDriver, getNeode } from './neo4j'
+
+const neode = getNeode()
+
+export const cleanDatabase = async (options = {}) => {
+  const { driver = getDriver() } = options
+  const session = driver.session()
+  try {
+    await session.writeTransaction(transaction => {
+      return transaction.run(
+        `
+          MATCH (everything)
+          DETACH DELETE everything
+        `,
+      )
+    })
+  } finally {
+    session.close()
+  }
+}
+
+Factory.define('category')
+  .attr('id', uuid)
+  .attr('icon', 'globe')
+  .attr('name', 'global-peace-nonviolence')
+  .after((buildObject, options) => {
+    return neode.create('Category', buildObject)
+  })
+
+Factory.define('badge')
+  .attr('type', 'crowdfunding')
+  .attr('status', 'permanent')
+  .after((buildObject, options) => {
+    return neode.create('Badge', buildObject)
+  })
+
+Factory.define('userWithoutEmailAddress')
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-02-04T09:29:16Z" title="Tuesday, February 4th 2020, 10:29:16 am +01:00">Feb 4, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,227 @@
+import uuid from 'uuid/v4'
+import faker from 'faker'
+import slugify from 'slug'
+import { hashSync } from 'bcryptjs'
+import { Factory } from 'rosie'
+import { getDriver, getNeode } from './neo4j'
+
+const neode = getNeode()
+
+export const cleanDatabase = async (options = {}) => {
+  const { driver = getDriver() } = options
+  const session = driver.session()
+  try {
+    await session.writeTransaction(transaction => {
+      return transaction.run(
+        `
+          MATCH (everything)
+          DETACH DELETE everything
+        `,
+      )
+    })
+  } finally {
+    session.close()
+  }
+}
+
+Factory.define('category')
+  .attr('id', uuid)
+  .attr('icon', 'globe')
+  .attr('name', 'global-peace-nonviolence')
+  .after((buildObject, options) => {
+    return neode.create('Category', buildObject)
+  })
+
+Factory.define('badge')
+  .attr('type', 'crowdfunding')
+  .attr('status', 'permanent')
+  .after((buildObject, options) => {
+    return neode.create('Badge', buildObject)
+  })
+
+Factory.define('userWithoutEmailAddress')
```


```suggestion
Factory.define('userWithFakerEmail')
```
not in love with this either, but how can we have a user without an email address?
and actually, if you look at it, this user does have an email address, it's just be chosen from `faker` instead of their choice

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-02-04T09:50:15Z" title="Tuesday, February 4th 2020, 10:50:15 am +01:00">Feb 4, 2020</time>_

---


oh, so looking at down further `userWithoutEmailAddress` has an email property, but it's not linked to that user, correct?

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2020-02-04T22:03:21Z" title="Tuesday, February 4th 2020, 11:03:21 pm +01:00">Feb 4, 2020</time>_

---


There is one test which needs a user without email address node. So I created two factories for the user, the `user` factory extending the `userWithoutEmailAddress` base factory.

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-02-10T11:47:01Z" title="Monday, February 10th 2020, 12:47:01 pm +01:00">Feb 10, 2020</time>_

---


I saw that, but you have `email` as a property, for some reason. This is confusing and should be removed, unless I haven't seen something.